### PR TITLE
Fix inaccuracy regarding drop of guard during unwinding

### PR DESCRIPTION
**Pull request by danielhenrymantilla**
Hey, thanks for the post! 🙏 

I've nonetheless wanted to chime in and suggest rewording a tiny bit the section around unwinding and lock poisoning 🙂

Indeed, a panic (or rather, the (stack) unwinding caused by one), does not prevent drop glue from running (in fact, the very reason the unwinding mechanism exists is precisely to be running drop glue!). On the other hand, you are right to point out the `Drop impl` of `MutexGuard` is indeed special insofar it does look at whether it is happening in an unwinding context, in which case it decides to unlock the mutex, but only after having left a "note"/flag in there for the record. Since the `.lock()` operation refuses to silently succeed when that flag has been set, the whole thing is called "poisoning".

I have also removed the "destructors always run" sentence, since code could incorrectly rely on this: memory cycles, or calling the `mem::forget` shorthand explicitly, do prevent destructors from running.

If you wanted to keep a sentence along these lines, then I would suggest to say something along the lines of:
> neither panicking nor `?` (nor other control flow interruptions such as `return`, `break`, and `continue`) prevents destructors from running [making it a very robust mechanism to avoid forgetting to release the mutex]

### CODE REVIEW

**Comment by hds** [comment]
@danielhenrymantilla thank you very much for your PR! As you may have noticed, it's the first one on this repo!

Just a heads up that I'm on holidays at the moment, so I won't merge it until I'm back at home and can publish the change to my website as well.

I just wanted you to know that this PR not getting merged isn't because I'm not planning on merging, it will just take a little bit.

**Comment by danielhenrymantilla** [comment]
No hurries! 🙂

File: content/posts/understanding-async-await-3.md [hunk 1]
```diff
@@ -1237,4 +1233,4 @@ A huge thank-you to [Conrad Ludgate](https://github.com/conradludgate) and [Pred
 
 Thanks to [sak96](https://sak96.github.io/) for reminding me that there is a lint for holding a guard across an await point.
 
-(in alphabetical order)
\ No newline at end of file
+(in alphabetical order)
```
Comment (hds) [inline content/posts/understanding-async-await-3.md:1236]:
```suggestion
(in alphabetical order)
```

**Review by hds — APPROVED** [review]
Thank you very much for your suggestions!

I've merged them as is, but will make a small change to split the new line into shorter lines before publishing to the web-site.

If you are happy for me to do so, I would like to credit you with your contribution in the **thanks** section. If this is fine, please let me know the name and a link you would like me to use.

Thanks!

**Comment by danielhenrymantilla** [comment]
Thanks! You can use [Daniel "Yandros" Henry-Mantilla](https://github.com/danielhenrymantilla) if you want 🙂

**Comment by hds** [comment]
Done. Thank you!
